torch/fx/experimental/Partitioner.py

```diff
@@ -0,0 +1,151 @@
+from torch.fx import Graph, GraphModule
+from torch.fx import Node
+from typing import Dict, List, Tuple
+"""a graph module may not fit into one backend.
+Partitioner class helps cut one graph into subgraphs (partitions),
+so that each partition could fit into different backends
+"""
+class Partition:
+    def __init__(self, partition_id: int) -> None:
+        self.parents: List[Partition] = []
+        self.children: List[Parttion] = []
+        """
+           One partion only has to fit into one device
+           'backend type' is the type of that device
+        """
+        self.partition_id = partition_id
+        """
+           self.node_list contains graph nodes that will be deployed in this partition
+           List[[node_index, node_name, node_op]]
+           with node_ndex, it is easy to track if any node is missing from all partitions
+           node_name is to show what this node does
+           node_op is used to find placeholders for input node of the partition
+        """
+        self.node_list: List[[Tuple[int, str, str]]] = []
+
+    def add_node(self, node_index: int, node_name: str, node_op: str) -> None:
+        """append a new node into the partition"""
+        self.node_list.append((node_index, node_name, node_op))
+
+    def add_parent(self, partition: 'Partition') -> None:
+        self.parents.append(partition)
+
+    def add_child(self, partition: 'Partition') -> None:
+        self.children.append(partition)
+
+    def get_input_nodes(self) -> [int]:
+        """input nodes are coming from two places:
+        placeholder or output from its parents output
+        return a list of node indexes
+        """
+        input_nodes: List[Tuple[int, str, str]] = []
+        for node_index, node_name, node_op in self.node_list:
+            if node_op == 'placeholder':
+                input_nodes.append((node_index, node_name, node_op))
+        for partition_id in self.parents:
+            input_nodes.append(partition.get_output_node())
+        return input_nodes
+
+    def get_output_node(self) -> int:
```

Comment:
Should be output nodes plural, It is very possible for a partition to have more than one output. Maybe not for the single partition case, but in the general case.